Remotion/Web/UnitTests/Core/UI/Controls/WebTabCollectionTest.cs

```diff
@@ -401,6 +401,34 @@ public void SelectInvisibleTab ()
         Throws.InvalidOperationException);
   }
 
+  [Test]
+  public void HideTabFollowedByInvisibleTab_WithFirstTabAsLastSelectable ()
+  {
+    _tabStrip.Tabs.Add(_tab0);
+    _tabStrip.Tabs.Add(_tab1);
+    _tabStrip.Tabs.Add(_tab2);
+    _tab1.IsSelected = true;
+    _tab2.IsVisible = false;
+    Assert.That(_tabStrip.SelectedTab, Is.SameAs(_tab1));
+    _tab1.IsVisible = false;
+
+    Assert.That(_tabStrip.SelectedTab, Is.SameAs(_tab0));
+  }
+
+  [Test]
+  public void HideTabFollowedByInvisibleTab_WithLastTabAsLastSelectable ()
+  {
+    _tabStrip.Tabs.Add(_tab0);
+    _tabStrip.Tabs.Add(_tab1);
+    _tabStrip.Tabs.Add(_tab2);
+    _tab0.IsSelected = true;
+    _tab1.IsVisible = false;
+    Assert.That(_tabStrip.SelectedTab, Is.SameAs(_tab0));
+    _tab0.IsVisible = false;
+
+    Assert.That(_tabStrip.SelectedTab, Is.SameAs(_tab2));
+  }
+
```

Comment:
Something from the tests to verify:
basically, the semantic is, to only select the first visible previous tab, if there is no visible next tab.

I think we need a test that shows that 
tab0.visbile=true
tab1.visibible=true
tab2.visible=false
tab3.visible=true
tab1.selected=true;
tab1.visible=false;
assert tab3.selected==true